river/optim/newton.py

```diff
@@ -45,10 +45,10 @@ def sherman_morrison(A_inv: dict, u: dict, v: dict) -> dict:
 
     den = 1 + utils.math.dot(utils.math.dotvecmat(u, A_inv), v)
 
-    for k, v in utils.math.matmul2d(
+    for k, val in utils.math.matmul2d(
```

Comment:
This rename resolves a Mypy warning, which came up because we were modifying the iterable inside a running loop.